core/client/src/main/java/alluxio/client/block/DelegatedUnderStoreBlockInStream.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the “License”). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.block;
+
+import alluxio.AlluxioURI;
+import alluxio.client.file.FileSystemContext;
+import alluxio.client.file.FileSystemWorkerClient;
+import alluxio.client.file.UnderFileSystemFileInStream;
+import alluxio.client.file.options.CloseUfsFileOptions;
+import alluxio.client.file.options.OpenUfsFileOptions;
+import alluxio.exception.AlluxioException;
+import alluxio.exception.ExceptionMessage;
+
+import java.io.IOException;
+
+/**
+ * Implementation of {@link UnderStoreBlockInStream} which accesses the under storage through an
+ * Alluxio worker. Note that the seek implementation of this class is not fail fast, and seeking
+ * past the end of the file will be successful until a subsequent read occurs.
+ */
+public class DelegatedUnderStoreBlockInStream extends UnderStoreBlockInStream {
```

Comment:
With DelegatedUnderStore, worker should call `ufs.connectFromWorker()` so that it can access secure HDFS.
Not sure where exactly we should add it, maybe in `OutputStreamAgent`, `InputStreamAgent`, after the UnderFileSystem object is created?